libtransact/src/context/mod.rs

```diff
@@ -19,15 +19,25 @@
 /// modify events, data, and state.
 pub type ContextId = [u8; 16];
 
+mod error;
 pub mod manager;
 
 use crate::receipts::Event;
 use std::fmt::Debug;
 use std::mem;
 use uuid::Uuid;
 
+pub use crate::context::error::ContextManagerError;
 use crate::receipts::StateChange;
 
+/// ContextManager functionality used by the Scheduler.
+pub trait ContextLifecycle<K, V, R>: Send {
+    /// Create a new Context, returning a unique ContextId.
+    fn create_context(&mut self, dependent_contexts: Vec<ContextId>, state_id: &str) -> ContextId;
+
+    fn drop_context(&mut self, context_id: &ContextId);
```

Comment:
I wonder if this method should "consume" the and convert it into the state changes for the receipt, combining the `get_transaction_receipt` with this function.  It seems like that the next stage in a context's life cycle.

Reply:
Not addressed, based on the discussion earlier